packages/app/src/backend/adapters/device-base-info/device-base-info.adapter.ts

```diff
@@ -12,42 +12,57 @@ import DeviceBaseInfoAdapter from "Backend/adapters/device-base-info/device-base
 import DeviceInfo from "Common/interfaces/device-info.interface"
 import CryptoFileService from "App/file-system/services/crypto-file-service/crypto-file-service"
 
-export class DeviceBaseInfo extends DeviceBaseInfoAdapter {
-  private token: string = CryptoFileService.createToken({
-    key: Math.random().toString(36).slice(-8),
+// TODO move those logic to DeviceModule.getDeviceInfo()
+let token: string
+
+const createToken = (): string => {
+  if (!token) {
+    token = CryptoFileService.createToken({
+      key: Math.random().toString(36).slice(-8),
+    })
+  }
+
+  return token
+}
+
+export const getDeviceInfoRequest = async (
```

Comment:
I'm not sure about it, but I think we can use it, just mark it as a temporary solution

Reply:
totally 😅

Reply:
[marked](https://github.com/mudita/mudita-center/pull/672/files/d3d93b2699876aa324be9b81dcf15ff77486193b#diff-61b588b77d268d4037efa7586e61062537287e82f4c72aac11c82408a5bd6b38R15)